opennms-doc/guide-development/src/asciidoc/text/dev-guide.adoc

```diff
@@ -0,0 +1,96 @@
+
+=== Installation
+
+The next chapter describes the full setup of your environment in order to meet the pre-requirements.
+Simply follow these instructions, they may vary depending on your Operating System.
+
+[source, shell]
+----
+# add OpenNMS as repository to install icmp and such
+echo "deb http://debian.opennms.org stable main" > /etc/apt/sources.list.d/opennms.list
+echo "deb-src http://debian.opennms.org stable main" >> /etc/apt/sources.list.d/opennms.list
+# Add pgp key
+wget -O - http://debian.opennms.org/OPENNMS-GPG-KEY | apt-key add -
+
+# overall update
+apt-get update
+# install stuff
+apt-get install -y software-properties-common wget git-core
+apt-get install -y tree tcpdump iftop htop slurm screen
+apt-get install -y zsh
+apt-get install -y nsis
+
+# install and configure PostgreSQL
+apt-get install -y postgresql
+echo "local   all             postgres                                peer" > /etc/postgresql/9.3/main/pg_hba.conf
+echo "local   all             all                                     peer" >> /etc/postgresql/9.3/main/pg_hba.conf
+echo "host    all             all             127.0.0.1/32            trust" >> /etc/postgresql/9.3/main/pg_hba.conf
+echo "host    all             all             ::1/128                 trust" >> /etc/postgresql/9.3/main/pg_hba.conf
+# install OpenNMS basic dependencies
+apt-get install -y maven
+apt-get install -y jicmp jicmp6
+apt-get install -y jrrd
+# fix locales
+echo 'LC_ALL="en_US.utf8"' >> /etc/environment
+locale-gen
+# set shell to use zsh
+chsh -s /bin/zsh $USER
+
+# clone opennms
+mkdir -p ~/dev/opennms
+git clone https://github.com/OpenNMS/opennms.git ~/dev/opennms
+----
+
+After this you should be able to build OpenNMS:
+
+[source, shell]
+----
+cd ~/dev/opennms
+./clean.pl
+./compile.pl -DskipTests
+./assemble.pl -p dir
+----
+
+After OpenNMS successfully built, please follow the instructions link:http://www.opennms.org/wiki/Installation:Source[here] under "Running OpenNMS", to start OpenNMS.
+
+=== Tooling
+We recommend the following toolset:
+
+ * Git
+ * Maven 3
+ * IDE: IntelliJ IDEA Ultimate
+ * DB-Tool: DBeaver or Postgres Admin
+ * Graphing: yEd
+ * Other: link:atom.io[www.atom.io]
+
+
+=== Useful links
+
+==== General
+
+ * https://www.github.com/OpenNMS/opennms: The source code hosted on GitHub
+ * http://wiki.opennms.org: Our Wiki, especially the start page is of interest. It points you in the right directions.
+ * http://issues.opennms.org: Our issue/bug tracker.
+ * https://github.com/opennms-forge/vagrant-opennms-dev: A vagrant box to setup a virtual box to build OpenNMS
+ * https://github.com/opennms-forge/vagrant-opennms: A vagrant box to setup a virtual box to run OpenNMS
+
+==== Installation / Setup
+* http://www.opennms.eu/docs/opennms-community-welcome-guide/0.0.5-SNAPSHOT/[]
+* http://www.opennms.org/wiki/Installation:Source[]
+* http://www.opennms.org/wiki/Developing_with_Git[]
+* http://www.opennms.org/wiki/Eclipse_and_OpenNMS[]
```

Comment:
We should also include the Idea and OpenNMS documentation: `http://www.opennms.org/wiki/IDEA_and_OpenNMS`